lib/pages/faq/service/question_provider.dart

```diff
@@ -0,0 +1,47 @@
+import 'package:acs_upb_mobile/generated/l10n.dart';
+import 'package:acs_upb_mobile/pages/faq/model/question.dart';
+import 'package:acs_upb_mobile/widgets/toast.dart';
+import 'package:cloud_firestore/cloud_firestore.dart';
+import 'package:flutter/cupertino.dart';
+
+class QuestionsProvider {
+  Future<QuerySnapshot> getDocuments(BuildContext context) {
+    try {
+      return Firestore().collection('faq').getDocuments();
+    } catch (e) {
+      print(e);
+      if (context != null) {
+        AppToast.show(S.of(context).errorSomethingWentWrong);
+      }
+      return null;
+    }
+  }
+
+  Future<QuerySnapshot> getPreviewDocuments(BuildContext context) {
+    try {
+      return Firestore().collection('faq').limit(2).getDocuments();
+    } catch (e) {
+      print(e);
+      if (context != null) {
+        AppToast.show(S.of(context).errorSomethingWentWrong);
+      }
+      return null;
+    }
+  }
```

Comment:
You can't return a `QuerySnapshot` from a Provider. The UI has to be completely separate from the database, it shouldn't know what a `QuerySnapshot` is (it's part of the Firebase API). It *does* know what a `Question` is though, so just make `getQuestions` take a (named) `context` parameter instead of a `QuerySnapshot` and return `Future<List<Question>>`. See `PersonProvider` for an example.

Reply:
You can also add an optional `limit` parameter to it for the preview, no need to have a separate method. (Side note: I didn't even know about the `limit()` call, very cool)